File: windows/StickyNotes.Windows/Models/Note.cs
namespace StickyNotes.Windows.Models;

public sealed class Note
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Title { get; set; } = "Untitled note";
    public string Content { get; set; } = string.Empty;
    public string Color { get; set; } = "Yellow";
    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.Now;

    public override string ToString() => Title;
}

File: macos/StickyNotes/Note.swift
import Foundation

struct Note: Identifiable, Codable, Hashable {
    var id = UUID()
    var title = "Untitled note"
    var content = ""
    var color = NoteColor.yellow
    var updatedAt = Date()
}

enum NoteColor: String, Codable, CaseIterable, Identifiable {
    case yellow, pink, blue, green, gray
    var id: Self { self }
    var label: String { rawValue.capitalized }
}
